spec/unit/indirector/request_spec.rb

```diff
@@ -521,4 +521,43 @@ def request(options = {})
       expect(request(:node => 'example.com', :ip => '127.0.0.1')).to be_remote
     end
   end
+
+  describe "failover" do
+    it "should use the provided failover host and port" do
+      Puppet.override(:server => 'myhost', :serverport => 666) do
+        req = Puppet::Indirector::Request.new('node', 'find', 'localhost', nil)
+        req.do_request() do |request|
+          expect(request.server).to eq('myhost')
+          expect(request.port).to eq(666)
+        end
+      end
+    end
+
+    it "should not use raw settings when failover fails" do
+      Puppet.override(:server => nil, :serverport => nil) do
+        req = Puppet::Indirector::Request.new('node', 'find', 'localhost', nil)
+        req.do_request() do |request|
+          expect(request.server).to be_nil
+          expect(request.port).to be_nil
```

Comment:
Can you also explicitly validate that the server and port settings aren't nil? ie explicitly assert we don't have a false negative?